### PR TITLE
[enh][broker] Support dynamic update between non-zero values of brokerPublisherThrottlingTickTimeMillis

### PR DESCRIPTION
**Pull request by Jason918**
### Motivation

Currently, config of `brokerPublisherThrottlingTickTimeMillis` is marked as dynamic, but only support update between zero and non-zero. It does not support like increase the value from 1 to 1000.

### Modifications

1. Add a class `PublishRateLimiterMonitor` to support tick interval updates.
2. Removed synchronized for `setupBrokerPublishRateLimiterMonitor`, synchronized on the big `BrokerService` object is not necessary and risky.
3. replace `awaitTermination` with `shutdownNow` to avoid long time block on metadata thread

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:
  - BrokerServiceTest#testPublishRateLimiterMonitor
  - BrokerServiceTest#testDynamicBrokerPublisherThrottlingTickTimeMillis

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Check the box below and label this PR (if you have committer privilege).

Need to update docs? 
  
- [x] `no-need-doc`

### CODE REVIEW

**Review by RobertIndie — COMMENTED** [review]
Overall looks good to me. Currently, the test only covers `PublishRateLimiterMonitor `. Could we add a test to verify dynamic update values of `brokerPublisherThrottlingTickTimeMillis` using the BrokerService to make sure the BrokerService works fine?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 1]
```diff
@@ -638,40 +638,66 @@ public synchronized void setupTopicPublishRateLimiterMonitor() {
...
+        }
+
+        synchronized void startOrUpdate(long tickTimeMs, Runnable checkTask, Runnable refreshTask) {
+            if (this.scheduler != null) {
+                // we have old task running.
+                if (this.tickTimeMs == tickTimeMs) {
+                    // tick time not changed.
+                    return;
+                }
+                stop();
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:670]: if (this.scheduler != null && this.tickTimeMs != tickTimeMs) {
          stop();
}

**Review by Technoboy- — APPROVED** [review]
LGTM

**Comment by Jason918** [comment]
> Overall looks good to me. Currently, the test only covers `PublishRateLimiterMonitor `. Could we add a test to verify dynamic update values of `brokerPublisherThrottlingTickTimeMillis` using the BrokerService to make sure the BrokerService works fine?

@RobertIndie 
Added unit test org.apache.pulsar.broker.service.BrokerServiceTest#testDynamicBrokerPublisherThrottlingTickTimeMillis PTAL.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java (hunk 1 above)
Comment (Jason918) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:671]: Not the same, no need to restart if tickTime does not changed.

**Review by RobertIndie — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 2]
```diff
@@ -638,40 +639,62 @@ public synchronized void setupTopicPublishRateLimiterMonitor() {
...
+            // schedule task that sums up publish-rate across all cnx on a topic ,
+            // and check the rate limit exceeded or not.
+            scheduler.scheduleAtFixedRate(safeRun(checkTask), tickTimeMs, tickTimeMs, TimeUnit.MILLISECONDS);
+            // schedule task that refreshes rate-limiting bucket
+            scheduler.scheduleAtFixedRate(safeRun(refreshTask), 1, 1, TimeUnit.SECONDS);
+            this.tickTimeMs = tickTimeMs;
+            this.refreshTask = refreshTask;
+        }
+
+        synchronized void stop() {
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:684]: Shutdown the broker also need to stop the monitor

Comment (Jason918) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:684]: Yes, this remains the same as previous logic, the inner scheduler will be shutdown, see Line 837.

**Comment by Jason918** [comment]
@codelipenghui PTAL

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**